docs/branching-and-tagging.md

```diff
@@ -0,0 +1,98 @@
+Branches are independent lineages of snapshots and point to the head of the lineage. 
+Branches and tags have a maximum reference age property which control when the reference to the snapshot itself should be expired.
+Branches have retention properties which define the minimum number of snapshots to retain on a branch as well as the maximum age of individual snapshots to retain on the branch. 
+These properties are used when the expireSnapshots procedure is run. 
+For details on the algorithm for expireSnapshots, refer to the [spec](../../../spec#snapshot-retention-policy).
+
+## Use Cases
+
+Branching and tagging can be used for handling GDPR requirements and retaining important historical snapshots for auditing. 
+For example, branching and tagging can be used to faciliate the following audit use case which requires different levels of retention. 
+
+1. Retain the latest 5 days of snapshots for 1 week. This can be achieved by setting a branch retention policy where 5 days worth of snapshots will be kept, and the branch reference itself will be retained for 1 week. 
+2. Retain 1 snapshot per week for 1 month. This can be achieved by tagging the weekly snapshot and setting the tag retention to be a month.
+3. Retain 1 snapshot per month for 6 months. This can be achieved by tagging the monthly snapshot and setting the tag retention to be 6 months.
+4. Retain 1 snapshot per year forever. This can be achieved by tagging the annual snapshot. The default retention for branches and tags is forever.
+
+Branches can also be used as part of data engineering workflows, for enabling experimental branches for testing and validating new jobs.
+See below for some examples of how branching and tagging can facilitate these use cases.
+
+### Historical Tags
+
+![Historical Tags](../img/historical-snapshot-tag.png)
+
+The above diagram shows an example of using tag to retain historical snapshots. 
+In this case a end of year tag is created for the final snapshot in the year, and the tagged snapshot has a retention of 5 years for compliance reasons. 
+The tags can be created with the following Spark DDLs
+
+```sql
+-- Specify the snapshot ID 3 and 5 for EOY-2022 and EOY-2023 respectively. Retain for 1825 days (5 years)
+ALTER TABLE prod.db.table CREATE TAG 'EOY-2022' AS OF VERSION 3 RETAIN 1825 DAYS
+ALTER TABLE prod.db.table CREATE TAG 'EOY-2023' AS OF VERSION 5 RETAIN 1825 DAYS
+```
+
+### Audit Branch
+
+![Audit Branch](../img/audit-branch.png)
+
+The above diagram shows an example of using an audit branch for validating a write workflow. 
+
+1. Writes are performed on a separate `audit-branch` independent from the main table history. 
+```scala
+-- WAP Branch write
+spark.conf().set(SparkSQLProperties.WAP_BRANCH, "audit");
```

Comment:
I think you an use https://spark.apache.org/docs/latest/sql-ref-syntax-aux-conf-mgmt-set.html to make it fully SQL